examples/sst2/model.py

```diff
@@ -0,0 +1,169 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""LSTM classifier model for SST-2."""
+
+import functools
+from typing import Any, Callable, Dict, Text
+
+import flax
+from flax import nn
+import jax
+import jax.numpy as jnp
+from jax import lax
+
+import numpy as np
+
+# pylint: disable=arguments-differ,too-many-arguments
+
+
+@functools.partial(jax.jit, static_argnums=(0, 1, 2, 3))
+def create_model(seed: int, batch_size: int, max_len: int,
+                 model_kwargs: Dict[Text, Any]):
+  """Instantiates a new model."""
+  model_def = TextClassifier.partial(train=False, **model_kwargs)
+  _, initial_params = model_def.init_by_shape(
+      jax.random.PRNGKey(seed),
+      [((batch_size, max_len), jnp.int32),
+       ((batch_size,), jnp.int32)])
+  model = nn.Model(model_def, initial_params)
+  return model
+
+
+def word_dropout(inputs: jnp.ndarray, rate: float = None, unk_idx: int = None):
+  """Replaces a fraction (rate) of inputs with <unk>."""
+  assert unk_idx is not None, 'Please provide unk_idx to word_dropout.'
+  mask = jax.random.bernoulli(nn.make_rng(), p=rate, shape=inputs.shape)
+  return jnp.where(mask, jnp.array([unk_idx]), inputs)
+
+
+class Embedding(nn.Module):
+  """Embedding Module."""
+
+  def apply(self,
+            inputs: jnp.ndarray,
+            num_embeddings: int,
+            features: int,
+            emb_init: Callable[...,
+                               np.ndarray] = nn.initializers.normal(stddev=0.1),
+            frozen: bool = False):
+    # inputs.shape = <int64>[batch_size, seq_length]
+    embedding = self.param('embedding', (num_embeddings, features), emb_init)
+    embed = jnp.take(embedding, inputs, axis=0)
+    if frozen:  # Keep the embeddings fixed at initial (pretrained) values.
+      embed = lax.stop_gradient(embed)
+    return embed
+
+
+class LSTM(nn.Module):
+  """LSTM encoder. Turns a sequence of vectors into a vector."""
+
+  def apply(self,
+            inputs: jnp.ndarray,
+            lengths: jnp.ndarray,
+            hidden_size: int = None):
+    # inputs.shape = <float32>[batch_size, seq_length, emb_size].
+    # lengths.shape = <int64>[batch_size,]
+    batch_size = inputs.shape[0]
+    carry = nn.LSTMCell.initialize_carry(
+        jax.random.PRNGKey(0), (batch_size,), hidden_size)
+    _, outputs = flax.jax_utils.scan_in_dim(
+        nn.LSTMCell.partial(name='lstm_cell'), carry, inputs, axis=1)
+    return outputs[jnp.arange(batch_size), jnp.maximum(0, lengths - 1), :]
+
+
+class MLP(nn.Module):
+  """A 2-layer MLP."""
+
+  def apply(self,
+            inputs: jnp.ndarray,
+            hidden_size: int = None,
+            output_size: int = None,
+            output_bias: bool = False,
+            dropout: float = None,
+            train: bool = None):
+    # inputs.shape = <float32>[batch_size, seq_length, hidden_size]
+    hidden = nn.Dense(inputs, hidden_size, name='hidden')
+    hidden = nn.tanh(hidden)
```

Comment:
No action needed but relu or elu lead often to better results and more numerical stability - no nan.

Reply:
I'll experiment with it and update if it indeed works better.